cluster/addons/calico-policy-controller/typha-horizontal-autoscaler-deployment.yaml

```diff
@@ -32,3 +32,8 @@ spec:
             limits:
               cpu: 10m
       serviceAccountName: typha-cpha
+      tolerations:
```

Comment:
I don't think we want to add these tolerations to the deployments. We need to only add these tolerations to the daemonsets that need to run on each node.

Reply:
These deployments are, however, critical for Calico to operate properly.

If there are no available nodes then Calico may not function properly. Are we OK with that?

Happy to remove if you agree.

Reply:
Yeah, I think we should remove these.

People usually don't taint all nodes. They usually taint some nodes, for example, when they want to reserve those nodes for special workloads. In such circumstances, we want pods that need to run on all nodes to be schedulable on those nods and so such pods need to have wildcard tolerations. These pods are mostly daemonsets (ex. kube-proxy, fluentd etc.). There are other important pods (ex. kube-dns) that don't need to run on all nodes and so they don't have wildcard tolerations.

So, yeah, only add these wildcard tolerations to the pods that need to run on all nodes and remove them from pods that don't need to run on all nodes.

Reply:
kk, pushed new changes.

Reply:
Thanks. One more comment, LGTM otherwise.